---
title: "inStrain Analysis"
author: "`r Sys.info()['user']`"
date: "`r Sys.Date()`"
output:
  html_document:
    code_folding: hide
    toc: true
    toc_float: true
    toc_depth: 4
    df_print: paged
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(dpi = 200)
knitr::opts_chunk$set(fig.align = "center")
knitr::opts_chunk$set(rows.print = 5)
knitr::opts_chunk$set(max.print = 100)

options(bitmapType='cairo')
```


# Introduction

Here I will prepare the inStrain output for the final paper figure.

Side note: All hashed items can be deleted before publishing on Git.
The original version of this (which I used for lab meeting is in the parent directory, called "V2")

# Setup

**Paths & Params**

```{r}
output_dir = '/ebio/abt3_projects/HUBIF_metagenomics/data/pipelines/llmgps/cophyverttrans/batch2_clusters1-3/instrain/n784/llmgps_output'
genomeWide_compare = file.path(output_dir, "/inStrain/output/inStrain_genomeWide_compare.tsv")
metadata = '/ebio/abt3_projects/HUBIF_metagenomics/data/metadata/final_metadata/combined_LFCurated_01.07.2020.txt'
gtdbtk = '/ebio/abt3_projects/HUBIF_metagenomics/data/pipelines/llg/cophyverttrans/batch2_clusters1-3/round3/llg_output/gtdbtk/gtdbtk_to_ncbi/gtdbtk_output_reformatted.tsv'

gtdbtk_to_NCBI = '/ebio/abt3_projects/HUBIF_metagenomics/data/pipelines/llg/cophyverttrans/batch2_clusters1-3/round3/llg_output/gtdbtk/gtdbtk_to_ncbi/gtdb_to_taxdump/output/taxonomy_map_summary.tsv'

#IDs_for_paper = '/ebio/abt3_projects/HUBIF_metagenomics/data/metadata/final_metadata/IDs_mom_infant_n386_2021LFcorrected2.txt'
IDs_for_paper = '/ebio/abt3_projects/HUBIF_metagenomics/data/metadata/cophy_vertrans_paper_ids/IDs_mom_infant_n386_2021LFcorrected2.txt'
```

**Libraries**

```{r lib load, message=FALSE}
library(dplyr)
library(tidyr)
library(ggplot2)
library(readr)
library(stringr)
library(ggpubr)
library(LeyLabRMisc)
#library(ape)
#library(vegan)
```

Loading data

```{r}
gtdbtk <- read_tsv(gtdbtk)

gtdbtk_to_NCBI <- read_tsv(gtdbtk_to_NCBI)
gtdbtk_to_NCBI <- gtdbtk_to_NCBI[-1,] #this removes the first row, which is an artifact of the dataframe before the python script was run

IDs_for_paper <- read_tsv(IDs_for_paper)
metadata <- read_tsv(metadata)

genomeWide_compare <- read_tsv(genomeWide_compare)
genomeWide_compare <- genomeWide_compare %>% 
                              mutate(name1 = str_replace_all(name1, ".bam", "")) %>% 
                              mutate(name2 = str_replace_all(name2, ".bam", "")) %>%
                              mutate(genome = str_replace_all(genome, ".fna", ""))

```

Parsing out the NCBI classification by phylum, genus, etc.
```{r}
gtdbtk_to_NCBI <- gtdbtk_to_NCBI %>% separate(lineage, sep = ";", into = taxonomy_levels(), remove = F)
gtdbtk_to_NCBI[is.na(gtdbtk_to_NCBI)] <- 'unclassified'
```


Combining GTDB-TK and NCBI results
```{r}
GTDBTk_NCBI <- full_join(gtdbtk, gtdbtk_to_NCBI, by = c("gtdbtk_mostgranular" = "gtdb_taxonomy"))
```

Adding GTDB-Tk and NCBI metadata to the inStrain output
```{r}
genomeWide_compare <- left_join(genomeWide_compare, GTDBTk_NCBI, by = c("genome" = "gtdbtk_user_genome"))
```

# Filtering the inStrain output by the IDs we are using for the study (this filters out 19 instances of `inStrain compare`)
```{r}
genomeWide_compare <- genomeWide_compare %>% subset(genomeWide_compare$name1 %in% IDs_for_paper$MomID |
                                                         genomeWide_compare$name2 %in% IDs_for_paper$MomID |
                                                         genomeWide_compare$name1 %in% IDs_for_paper$InfantID |
                                                         genomeWide_compare$name2 %in% IDs_for_paper$InfantID)

#Writing file for Hagay, so he can make the final figure (1/2)
write_tsv(x=GTDBTk_NCBI, file="/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/instrain_output_for_figure/SRGs_GTDBTk_NCBI.tsv")
```

# Analysis

```{r}
metadata_cols <- c("ParticipantID","Infant1ID","Infant2ID","Infant3ID","InfantAdult","MotherID","SamplingLocation_2","SamplingLocation_1", "CurrCountry", "Age_months")
metadata_cols <- metadata %>% select(metadata_cols)

colnames(metadata_cols) <- paste("name1",colnames(metadata_cols),sep="_")

add_metadata_name1 <- . %>% left_join(metadata_cols, by = c("name1" = "name1_ParticipantID"))

genomeWide_compare <- genomeWide_compare %>% add_metadata_name1

colnames(metadata_cols) <- colnames(metadata_cols) %>% str_replace_all("name1_", "")
colnames(metadata_cols) <- paste("name2",colnames(metadata_cols),sep="_")
add_metadata_name2 <- . %>% left_join(metadata_cols, by = c("name2" = "name2_ParticipantID"))

genomeWide_compare <- genomeWide_compare %>% add_metadata_name2
```
 
Adding whether each pair is a MIP or not
```{r}
make_new_columns <- . %>% 
                              mutate(MIP = case_when(
                                              name1_MotherID == name1  ~ '1',
                                              name2_MotherID == name1  ~ '1',
                                              name1_MotherID == name2  ~ '1',
                                              name2_MotherID == name2  ~ '1',
                                              TRUE ~ "0")) %>%
                              mutate(MI = case_when(
                                              name1_InfantAdult == "Adult" & name2_InfantAdult == "Infant"  ~ '1',
                                              name1_InfantAdult == "Infant" & name1_InfantAdult == "Adult"  ~ '1',
                                              TRUE ~ "0")) %>%
                              mutate(SamplingLocation_2 = case_when(
                                              name1_SamplingLocation_2 == name2_SamplingLocation_2  ~ '1',
                                              TRUE ~ "0")) %>%
                              mutate(SamplingLocation_1 = case_when(
                                              name1_SamplingLocation_1 == name2_SamplingLocation_1  ~ '1',
                                              TRUE ~ "0")) %>%
                              mutate(CurrCountry = case_when(
                                              name1_CurrCountry == name2_CurrCountry ~ '1',
                                              TRUE ~ "0")) %>%
                              mutate(Level = case_when(
                                              MIP == 1 ~ 'MIP',
                                              SamplingLocation_2 == 1 ~ 'SamplingLocation_2', 
                                              SamplingLocation_1 == 1 ~ 'SamplingLocation_1',
                                              CurrCountry == 1 ~ 'CurrCountry',
                                              TRUE ~ "Inter-continental")) %>%
                              mutate(Level_new = case_when(
                                              MIP == 1 ~ 'MIP',
                                              SamplingLocation_2 == 1 ~ 'Town/Village', 
                                              SamplingLocation_1 == 1 ~ 'Region',
                                              CurrCountry == 1 ~ 'Country',
                                              TRUE ~ "International"))

genomeWide_compare <- genomeWide_compare %>% make_new_columns
comp50 <- subset(genomeWide_compare, percent_compared >= 0.50)
comp50_popANI99999 <- subset(comp50, popANI >= 0.99999)
```

Writing enhanced inStrain output file for Hagay to make final figure (2/2)
```{r}
#write_tsv(x=comp50, file="/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/instrain_output_for_figure/instrain_comp50output_wmetadata.tsv")

test change to file, testing ... testing test123 testing123
```


#Part I: The SSEs
```{r}
# Only 148 of the 118587 comparisons were within MIPs
comp50 %>% count(MIP, sort = TRUE)

# 158; 65 were within MIPs and 93 were between unpaired MIPS or mother-mother/child-child
comp50_popANI99999 %>% count(MIP, sort = TRUE)

# 102 SSE in Gabon; 36 (35%) were within MCPs; 53 (52%) were at town/village; 13 (13%) were at region lvl
comp50_popANI99999 %>% subset(name1_CurrCountry == "Gabon" | name2_CurrCountry == "Gabon") %>% count(Level, sort = TRUE)
# 38 SSE in Vietnam; 11 (29%) were within MCPs; 24 (63%) were at town/village; 1 (3%) was at region lvl 
comp50_popANI99999 %>% subset(name1_CurrCountry == "Vietnam" | name2_CurrCountry == "Vietnam") %>% count(Level, sort = TRUE)
# 18 SSE in Germany; 100% were within MCPs
comp50_popANI99999 %>% subset(name1_CurrCountry == "Germany" | name2_CurrCountry == "Germany") %>% count(Level, sort = TRUE)

HTEs <- comp50_popANI99999 %>% subset(MIP == 0) %>%
                               count(ncbi_taxonomy, name1_SamplingLocation_1, sort = TRUE)
VTEs <- comp50_popANI99999 %>% subset(MIP == 1) %>%
                              count(ncbi_taxonomy, name1_SamplingLocation_1, sort = TRUE)

comp50_popANI99999 %>% subset(MIP == 1) %>%
                               count(name1_SamplingLocation_1, sort = TRUE)

# 36 of the VTEs were in Gabon, 18 in Germany and 11 in Vietnam
top_taxa_VTEs_ncbi <- count(VTEs, ncbi_taxonomy) #top_taxa_VTEs_srg <- count(VTEs, genome)
top_taxa_comp50_srg <- count(comp50, genome)

phylum_sses <- count(comp50_popANI99999, Phylum, sort = TRUE)

comp50_popANI99999 %>% 
    count(Phylum, sort = TRUE) %>% 
    ggplot(aes(x=reorder(Phylum, -n), y = n))+ # add , fill = name1_SamplingLocation_1)?xt
    geom_bar(stat="identity", fill="steelblue")+
    labs(title="Phyla of Shared Strains",x="Phylum", y = "Number of Strain-Sharing Events")+
    theme_light()+
    theme(axis.text.x = element_text(angle = 30, hjust=0.90))+
    theme(plot.title = element_text(hjust = 0.5))
ggsave('/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/earlysept_analysis_figures/sharedstrain_phyla.png')

comp50_popANI99999 %>%
    subset(MIP == 1) %>% 
    count(Phylum, sort = TRUE) %>% 
    ggplot(aes(x=reorder(Phylum, -n), y = n))+ # add , fill = name1_SamplingLocation_1)?xt
    geom_bar(stat="identity", fill="steelblue")+
    labs(title="Phyla of Shared Strains: MCPs",x="Phylum", y = "Number of Strain-Sharing Events")+
    theme_light()+
    theme(axis.text.x = element_text(angle = 30, hjust=0.90))+
    theme(plot.title = element_text(hjust = 0.5))
ggsave('/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/earlysept_analysis_figures/sharedstrain_phylaMCPs.png')
```

```{r}
ggplot(comp50 %>% filter(!is.na(Level)), aes(x=MIP, y=popANI)) +
  labs(title="Degree of SRG Relatedness", x="MIP", y = "popANI")+
  theme_light()+
  geom_violin()+
  theme(plot.title = element_text(hjust = 0.5))+
  geom_boxplot(width=0.1)

comp50 %>%
  subset(name1_Age_months > 0 | name2_Age_months > 0) %>%
  mutate(name1_Age_months = if_else(is.na(name1_Age_months), 0, name1_Age_months)) %>%
  mutate(name2_Age_months = if_else(is.na(name2_Age_months), 0, name2_Age_months)) %>%
  mutate(Infant_AgeMonths = name1_Age_months + name2_Age_months) %>% 
    ggplot(aes(x=Infant_AgeMonths, y=popANI, color = name1_CurrCountry)) +
      labs(title="Strain Relatedness Over Infants' Lives", x="Infant Age (Months)", y = "popANI")+
      geom_point()+
      theme_light()+
      geom_smooth(method=lm)+
      theme(plot.title = element_text(hjust = 0.5))
ggsave('/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/septlabmeetingfigs/comp50_popANI_infantage.png')

comp50 %>%
  subset(MIP == 1) %>% 
  subset(name1_Age_months > 0 | name2_Age_months > 0) %>%
  mutate(name1_Age_months = if_else(is.na(name1_Age_months), 0, name1_Age_months)) %>%
  mutate(name2_Age_months = if_else(is.na(name2_Age_months), 0, name2_Age_months)) %>%
  mutate(Infant_AgeMonths = name1_Age_months + name2_Age_months) %>% 
    ggplot(aes(x=Infant_AgeMonths, y=popANI, color = name1_CurrCountry)) +
      labs(title="Strain-Relatedness Over Infants' Lives: MCPs", x="Infant Age (Months)", y = "popANI")+
      geom_point()+
      theme_light()+
      theme(plot.title = element_text(hjust = 0.5))+
      geom_smooth(method=lm)
ggsave('/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/septlabmeetingfigs/wMCP_popANI_infantage.png')


comp50 %>%
  subset(name1_CurrCountry == "Vietnam") %>%
  subset(name1_Age_months > 0 | name2_Age_months > 0) %>%
  mutate(name1_Age_months = if_else(is.na(name1_Age_months), 0, name1_Age_months)) %>%
  mutate(name2_Age_months = if_else(is.na(name2_Age_months), 0, name2_Age_months)) %>%
  mutate(Infant_AgeMonths = name1_Age_months + name2_Age_months) %>% 
    ggplot(aes(x=Infant_AgeMonths, y=popANI)) +
      labs(title="Strain-Relatedness Over Infants' Lives: Vietnam Infants", x="Infant Age", y = "popANI")+
      geom_point()+
      stat_regline_equation(label.x=45, label.y=0.968) +
      stat_cor(aes(label=..rr.label..), label.x=45, label.y=0.963) +
      theme_light()+
      geom_smooth(method=lm)+
      theme(plot.title = element_text(hjust = 0.5))
ggsave('/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/septlabmeetingfigs/comp50_popANI_infantageViet.png')

comp50 %>%
  subset(MIP == 1) %>% 
  subset(name1_CurrCountry == "Vietnam") %>%
  subset(name1_Age_months > 0 | name2_Age_months > 0) %>%
  mutate(name1_Age_months = if_else(is.na(name1_Age_months), 0, name1_Age_months)) %>%
  mutate(name2_Age_months = if_else(is.na(name2_Age_months), 0, name2_Age_months)) %>%
  mutate(Infant_AgeMonths = name1_Age_months + name2_Age_months) %>% 
    ggplot(aes(x=Infant_AgeMonths, y=popANI)) +
      labs(title="Strain-Relatedness Over Infants' Lives: Vietnam Infants", x="Infant Age", y = "popANI")+
      geom_point()+
      stat_regline_equation(label.x=15, label.y=0.98) +
      stat_cor(aes(label=..rr.label..), label.x=15, label.y=0.975) +
      theme_light()+
      geom_smooth(method=lm)+
      theme(plot.title = element_text(hjust = 0.5))
ggsave('/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/septlabmeetingfigs/wMCP_popANI_infantageViet.png')
```



```{r}
cols <- c("p__actinobacteria" = "red", "p__bacteroidetes" = "blue", "p__firmicutes" = "green", "p__proteobacteria" = "orange", "p__spirochaetes" = "pink", "p__tenericutes" = "yellow", "unclassified" = "purple")

#Interrogating the SSEs
gabon_SS <- comp50 %>%
  subset(name1_CurrCountry == "Gabon" | name2_CurrCountry == "Gabon") %>%
  subset(popANI >= 0.99999) %>%
  arrange(Level_new) %>%
  mutate(Level_new = factor(Level_new, levels=c("MIP", "Town/Village", "Region", "Country", "International"))) %>%
  ggplot(aes(x=Level_new, y=popANI)) +
  labs(title="Strain Sharing Events in Gabon", x="Level", y = "popANI")+
  geom_boxplot(width=0.3)+
  theme_bw()+
  scale_colour_manual(
      values = cols,
      breaks = c("p__actinobacteria", "p__bacteroidetes", "p__firmicutes", "p__proteobacteria", "p__spirochaetes","p__tenericutes", "unclassified"),
      labels = c("Actinobacteria", "Bacteroidetes", "Firmicutes", "Proteobacteria", "Spirochaetes","Tenericutes", "Unclassified")
    )+
  scale_y_continuous(limit = c(0.9999900, 1.0000000))+
  theme(plot.title = element_text(hjust = 0.5, face="bold"))+
  theme(axis.text=element_text(size=11))+
  scale_x_discrete(name = "",
                   label = c("MIP", "Town/Village", "Region", "Country", "International"),
                   drop = FALSE)+
  geom_jitter(aes(color=Phylum))
#ggsave('/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/earlysept_analysis_figures/gabon_strainsharing.png')

gabon_SS

vietnam_SS <- comp50 %>%
  subset(name1_CurrCountry == "Vietnam" | name2_CurrCountry == "Vietnam") %>%
  subset(popANI >= 0.99999) %>%
  arrange(Level_new) %>%
  mutate(Level_new = factor(Level_new, levels=c("MIP", "Town/Village", "Region", "Country", "International"))) %>%
  ggplot(aes(x=Level_new, y=popANI)) +
  labs(title="Strain Sharing Events in Vietnam", x="Level", y = "popANI")+
  geom_boxplot(width=0.3)+
  scale_colour_manual(values = cols) +#scale_color_brewer(palette="Set1")+
  theme_bw()+
  scale_colour_manual(
      values = cols,
      breaks = c("p__actinobacteria", "p__bacteroidetes", "p__firmicutes", "p__proteobacteria", "p__spirochaetes","p__tenericutes", "unclassified"),
      labels = c("Actinobacteria", "Bacteroidetes", "Firmicutes", "Proteobacteria", "Spirochaetes","Tenericutes", "Unclassified")
    )+
  scale_y_continuous(limit = c(0.9999900, 1.0000000))+
  theme(plot.title = element_text(hjust = 0.5, face="bold"))+
  theme(axis.text=element_text(size=11))+
  scale_x_discrete(name = "",
                   label = c("MIP", "Town/Village", "Region", "Country", "International"),
                   drop = FALSE)+
  geom_jitter(aes(color=Phylum))
#ggsave('/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/earlysept_analysis_figures/vietnam_strainsharing.png')

germany_SS <- comp50 %>%
  subset(name1_CurrCountry == "Germany" | name2_CurrCountry == "Germany") %>%
  subset(popANI >= 0.99999) %>%
  arrange(Level_new) %>%
  mutate(Level_new = factor(Level_new, levels=c("MIP", "Town/Village", "Region", "Country", "International"))) %>%
  ggplot(aes(x=Level_new, y=popANI)) +
  labs(title="Strain Sharing Events in Germany", x="Level", y = "popANI")+
  geom_boxplot(width=0.3)+
  scale_colour_manual(values = cols) +#scale_color_brewer(palette="Set1")+
  theme_bw()+
  scale_colour_manual(
      values = cols,
      breaks = c("p__actinobacteria", "p__bacteroidetes", "p__firmicutes", "p__proteobacteria", "p__spirochaetes","p__tenericutes", "unclassified"),
      labels = c("Actinobacteria", "Bacteroidetes", "Firmicutes", "Proteobacteria", "Spirochaetes","Tenericutes", "Unclassified")
    )+
  scale_y_continuous(limit = c(0.9999900, 1.0000000))+
  theme(plot.title = element_text(hjust = 0.5, face="bold"))+
  theme(axis.text=element_text(size=11))+
  scale_x_discrete(name = "",
                   label = c("MIP", "Town/Village", "Region", "Country", "International"),
                   drop = FALSE)+
  geom_jitter(aes(color=Phylum))
#ggsave('/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/earlysept_analysis_figures/germany_strainsharing.png')

# this didn't work so i combined them manually
#ggarrange(gabon_SS, vietnam_SS,  common.legend = TRUE)
#ggsave('/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/earlysept_analysis_figures/3countries_strainsharing.png')
```


Running an ANOVA + Tukey HSD on the intra-country SSE differences by different geographic level
```{r}
anova_df <- comp50_popANI99999 %>% subset(name1_CurrCountry == "Gabon" | name2_CurrCountry == "Gabon")
# Compute the analysis of variance
res.aov <- aov(popANI ~ Level, data = anova_df)
# Summary of the analysis
summary(res.aov)
TukeyHSD(res.aov)

anova_df <- comp50_popANI99999 %>% subset(name1_CurrCountry == "Vietnam" | name2_CurrCountry == "Vietnam")
# Compute the analysis of variance
res.aov <- aov(popANI ~ Level, data = anova_df)
# Summary of the analysis
summary(res.aov)
TukeyHSD(res.aov)
```

#Part II: The inStrain comparisons with popANI <= 99.999%

```{r}
gabon_SS <- comp50 %>%
  subset(name1_CurrCountry == "Gabon" | name2_CurrCountry == "Gabon") %>%
  subset(popANI >= 0.999) %>%
  arrange(Level_new) %>%
  mutate(Level_new = factor(Level_new, levels=c("MIP", "Town/Village", "Region", "Country", "International"))) %>%
  ggplot(aes(x=Level_new, y=popANI)) +
  labs(title="popANI Cutoff Reduced to 99.9% (Gabon)", x="Level", y = "popANI")+
  geom_boxplot(width=0.3)+
  scale_colour_manual(values = cols) +#scale_color_brewer(palette="Set1")+
  theme_bw()+
  scale_colour_manual(
      values = cols,
      breaks = c("p__actinobacteria", "p__bacteroidetes", "p__firmicutes", "p__proteobacteria", "p__spirochaetes","p__tenericutes", "unclassified"),
      labels = c("Actinobacteria", "Bacteroidetes", "Firmicutes", "Proteobacteria", "Spirochaetes","Tenericutes", "Unclassified")
    )+
  theme(plot.title = element_text(hjust = 0.5, face="bold"))+
  theme(axis.text=element_text(size=11))+
  scale_x_discrete(name = "",
                   label = c("MIP", "Town/Village", "Region", "Country", "International"),
                   drop = FALSE)+
  geom_jitter(aes(color=Phylum))
#ggsave('/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/earlysept_analysis_figures/gabon_relaxedpopANI.png')

gabon_SS <- comp50 %>%
  subset(name1_CurrCountry == "Gabon" | name2_CurrCountry == "Gabon") %>%
  subset(popANI >= 0.999) %>%
  arrange(Level_new) %>%
  mutate(Level_new = factor(Level_new, levels=c("MIP", "Town/Village", "Region", "Country", "International"))) %>%
  ggplot(aes(x=Level_new, y=popANI)) +
  labs(title="popANI Cutoff Reduced to 97% (Gabon)", x="Level", y = "popANI")+
  geom_boxplot(width=0.3)+
  scale_color_brewer(palette="Set1")+
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5, face="bold"))+
  theme(axis.text=element_text(size=11))+
  scale_x_discrete(name = "",
                   label = c("MIP", "Town/Village", "Region", "Country", "International"),
                   drop = FALSE)+
  geom_jitter(aes(color=Phylum))
#ggsave('/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/earlysept_analysis_figures/gabon_999popANI.png')
```


There are 377 unique SRGs 
```{r}
unique_srgs_gtdbtk <- unique(comp50$genome)
unique_srgs_ncbi <- unique(comp50$ncbi_taxonomy)

length(unique_srgs)

top_taxa_50comp_gtdbtk <- count(comp50, genome)
top_taxa_50comp_ncbi <- count(comp50, ncbi_taxonomy)

top_taxa <- left_join(top_taxa_50comp_ncbi, gtdbtk, by = c("ncbi_taxonomy" = "gtdbtk_user_genome"))

top_3taxa <- subset(top_taxa, n > 1000)
top_3taxa_list <- top_3taxa$ncbi_taxonomy

comp50_test <- subset(comp50, ncbi_taxonomy %in% top_3taxa_list)
```

```{r}
ggplot(comp50_test, aes(x=popANI, y=ncbi_taxonomy), fill=MIP) +
  labs(title="7 Most Prevelant SRGs",x="popANI", y = "SRG")+
  theme_light()+
  geom_boxplot(aes(colour = MIP))+
  theme(plot.title = element_text(hjust = 0.5))
```
^ This plot shows how not all SRGs have >=0 MIPs that share them. We need to filter those out...

There are 55 SRGs that include at least 1 MIP-sharing
```{r}
comp50_noAAsIIs <- comp50 %>% filter(!(name1_InfantAdult == name2_InfantAdult))

taxa_wMIPs <- subset(comp50_noAAsIIs, MIP == 1)

unique_srgs_wMIPs <- unique(taxa_wMIPs$genome)
unique_ncbiclass_wMIPs <- unique(taxa_wMIPs$ncbi_taxonomy)

top_srgs_wMIPs <- count(taxa_wMIPs, genome)
top_ncbiclass_wMIPs <- count(taxa_wMIPs, ncbi_taxonomy)

top_srgs_wMIPs <- left_join(top_srgs_wMIPs, gtdbtk, by = c("genome" = "gtdbtk_user_genome"))
#top_ncbiclass_wMIPs <- left_join(top_ncbiclass_wMIPs, GTDBTk_NCBI, by = c("ncbi_taxonomy" = "ncbi_taxonomy"))

top_srgs_wMIPs <- subset(top_srgs_wMIPs, n > 4)
top_taxa_wMIPs_list <- top_srgs_wMIPs$genome

top_ncbiclass_wMIPs <- subset(top_ncbiclass_wMIPs, n > 3)
top_ncbiclass_wMIPs_list <- top_ncbiclass_wMIPs$ncbi_taxonomy

comp50_wMIPs_gtdbtk <- subset(comp50, genome == top_taxa_wMIPs_list)
comp50_wMIPs_ncbi <- subset(comp50, ncbi_taxonomy == top_ncbiclass_wMIPs_list)
```

```{r}
ggplot(comp50_wMIPs_gtdbtk, aes(x=popANI, y=gtdbtk_species), fill=MIP) +
  labs(title="SRG Sharing (GTDB-Tk)",x="popANI", y = "SRG")+
  theme_light()+
  geom_boxplot(aes(colour = MIP))+
  theme(plot.title = element_text(hjust = 0.5))
#ggsave('/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/labmeeting_analysis_figures/SSG_sharing_50perccompl.png')
```

```{r}
ggplot(comp50_wMIPs_ncbi, aes(x=popANI, y=ncbi_taxonomy), fill=MIP) +
  labs(title="NCBI Taxon Relatedness",x="popANI", y = "SRG")+
  theme_light()+
  geom_boxplot(aes(colour = MIP))+
  theme(plot.title = element_text(hjust = 0.5))
```
 
Part II: Expanding outside of the strict definition of recent transmission event (i.e., where popANI < 99.999%)
```{r}
#friday september 3rd, midnight
ggplot(comp50, aes(x=MIP, y=popANI)) +
  labs(title="Degree of SRG Relatedness", x="MCP", y = "popANI")+
  theme_light()+
  geom_violin()+
  theme(plot.title = element_text(hjust = 0.5))+
  geom_boxplot(width=0.1)
ggsave('/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/earlysept_analysis_figures/MCP_srgrelatedness.png')
```
```{r}
comp50 %>%
  subset(name1_CurrCountry == "Gabon" | name2_CurrCountry == "Gabon") %>%
  arrange(Level_new) %>%
  mutate(Level_new = factor(Level_new, levels=c("MIP", "Town/Village", "Region", "Country", "International"))) %>%
  ggplot(aes(x=Level_new, y=popANI)) +
  labs(title="All Comparisons - Gabon", x="Level", y = "popANI")+
  theme_light()+
  geom_violin()+
  scale_x_discrete(name = "",
                   label = c("MIP", "Town/Village", "Region", "Country", "International"),
                   drop = FALSE)+
  geom_boxplot(width=0.1)
ggsave('/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/earlysept_analysis_figures/gabon_comp50.png')

comp50 %>%
  subset(name1_CurrCountry == "Vietnam" | name2_CurrCountry == "Vietnam") %>%
  arrange(Level_new) %>%
  mutate(Level_new = factor(Level_new, levels=c("MIP", "Town/Village", "Region", "Country", "International"))) %>%
  ggplot(aes(x=Level_new, y=popANI)) +
  labs(title="All Comparisons - Vietnam", x="Level", y = "popANI")+
  theme_light()+
  geom_violin()+
  scale_x_discrete(name = "",
                   label = c("MIP", "Town/Village", "Region", "Country", "International"),
                   drop = FALSE)+
  geom_boxplot(width=0.1)
ggsave('/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/earlysept_analysis_figures/vietnam_comp50.png')

comp50 %>%
  subset(name1_CurrCountry == "Germany" | name2_CurrCountry == "Germany") %>%
  arrange(Level_new) %>%
  mutate(Level_new = factor(Level_new, levels=c("MIP", "Town/Village", "Region", "Country", "International"))) %>%
  ggplot(aes(x=Level_new, y=popANI)) +
  labs(title="All Comparisons - Germany", x="Level", y = "popANI")+
  theme_light()+
  geom_violin()+
  scale_x_discrete(name = "",
                   label = c("MIP", "Town/Village", "Region", "Country", "International"),
                   drop = FALSE)+
  geom_boxplot(width=0.1)
ggsave('/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/earlysept_analysis_figures/germany_comp50.png')

comp50 %>%
  arrange(Level_new) %>%
  mutate(Level_new = factor(Level_new, levels=c("MIP", "Town/Village", "Region", "Country", "International"))) %>%
  ggplot(aes(x=Level_new, y=popANI)) +
  labs(title="All Comparisons; All Countries", x="Level", y = "popANI")+
  theme_light()+
  geom_violin()+
  scale_x_discrete(name = "",
                   label = c("MIP", "Town/Village", "Region", "Country", "International"),
                   drop = FALSE)+
  geom_boxplot(width=0.1)
ggsave('/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/earlysept_analysis_figures/allcountries_comp50.png')
```

```{r}
test <- metadata %>%
  subset(SamplingLocation_1 == "Field") %>%
  select(SamplingLocation_2) %>%
  dplyr::distinct()
  

dplyr::distinct(test)
```


# Columns for Hagay (conversation notes)
Sample1, Sample2, popANI, MIP (yes/no), species name


# Summary



***

# SessionInfo

```{r sessionInfo}
sessionInfo()
```
