---
title: "inStrain Analysis"
author: "Liam Fitzstevens & Kelsey Huus"
date: "17.09.21"
output:
  html_document:
    code_folding: hide
    toc: true
    toc_float: true
    toc_depth: 4
    df_print: paged
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(dpi = 200)
knitr::opts_chunk$set(fig.align = "center")
knitr::opts_chunk$set(rows.print = 5)
knitr::opts_chunk$set(max.print = 100)

options(bitmapType='cairo')
```

# Introduction

This script generates figure 4A, along with some additional overview information about the strain sharing events.

# Setup

**Paths & Data Loading**

```{r}
comp50 = '/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/final_for_paper/codiversification/strain_sharing/files/comp50_wmetadata_wNCBI.tsv'
```

**Libraries**

```{r lib load, message=FALSE}
library(dplyr)
library(tidyr)
library(RColorBrewer)
library(ggplot2)
library(readr)
library(stringr)
library(LeyLabRMisc)
```

# Analysis

```{r}
comp50 <- read_tsv(comp50)
comp50_popANI99999 <- comp50 %>% subset(popANI >= 0.99999)
```

```{r}
# Only 148 of the 118587 comparisons were within MIPs
comp50 %>% count(MIP, sort = TRUE)

# 158; 65 were within MIPs and 93 were between unpaired MIPS or mother-mother/child-child
comp50_popANI99999 %>% count(MIP, sort = TRUE)

# 102 SSE in Gabon; 36 (35%) were within MCPs; 53 (52%) were at town/village; 13 (13%) were at region lvl
comp50_popANI99999 %>% subset(name1_CurrCountry == "Gabon" | name2_CurrCountry == "Gabon") %>% count(Level, sort = TRUE)
# 38 SSE in Vietnam; 11 (29%) were within MCPs; 24 (63%) were at town/village; 1 (3%) was at region lvl 
comp50_popANI99999 %>% subset(name1_CurrCountry == "Vietnam" | name2_CurrCountry == "Vietnam") %>% count(Level, sort = TRUE)
# 18 SSE in Germany; 100% were within MCPs
comp50_popANI99999 %>% subset(name1_CurrCountry == "Germany" | name2_CurrCountry == "Germany") %>% count(Level, sort = TRUE)

comp50_popANI99999 %>% subset(MIP == 0) %>%
                               count(ncbi_taxonomy, name1_SamplingLocation_1, sort = TRUE)
comp50_popANI99999 %>% subset(MIP == 1) %>%
                              count(ncbi_taxonomy, name1_SamplingLocation_1, sort = TRUE)

comp50_popANI99999 %>% subset(MIP == 1) %>%
                               count(name1_SamplingLocation_1, sort = TRUE)

# 36 of the VTEs were in Gabon, 18 in Germany and 11 in Vietnam
top_taxa_comp50_srg <- count(comp50, genome)

phylum_sses <- count(comp50_popANI99999, Phylum, sort = TRUE)
```
Figure 4A
```{r}
cols <- c("p__actinobacteria" = "#660099", "p__bacteroidetes" = "#FF0000", "p__firmicutes" = "#339933", "p__proteobacteria" = "#0000FF", "p__spirochaetes" = "#FF9900", "p__tenericutes" = "#744700", "unclassified" = "#0099FF")

pdf("/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/final_for_paper/codiversification/strain_sharing/files/strainsharing_Gabon.pdf", width=6.75, height=5.75)
gabon_SS <- comp50 %>%
  subset(name1_CurrCountry == "Gabon" | name2_CurrCountry == "Gabon") %>%
  subset(popANI >= 0.99999) %>%
  arrange(Level_new) %>%
  mutate(Level_new = factor(Level_new, levels=c("MIP", "Town/Village", "Region", "Country", "International"))) %>%
  ggplot(aes(x=Level_new, y=popANI)) +
    labs(title="Gabon", x="Level", y = "popANI")+
    geom_boxplot(outlier.shape=NA, outlier.size=NA) +
    geom_jitter(size=3, width=0.3, height=0,aes(color=Phylum))+
    scale_y_continuous(limit = c(0.9999900, 1.0000000))+
    theme(plot.title = element_text(hjust = 0.5),
          text = element_text(family = "Helvetica"),
          axis.text=element_text(size=11),
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          panel.background = element_blank(), axis.line = element_line(colour = "black"),
          axis.title.x = element_text(margin = margin(t = 10)))+
    scale_x_discrete(name = "",
                   label = c("Mother-Child", "Town", "Region", "Country", "International"),
                   drop = FALSE)+
    scale_colour_manual(
        values = cols,
        breaks = c("p__actinobacteria", "p__bacteroidetes", "p__firmicutes", "p__proteobacteria", "p__spirochaetes","p__tenericutes", "unclassified"),
        labels = c("Actinobacteria", "Bacteroidetes", "Firmicutes", "Proteobacteria", "Spirochaetes","Tenericutes", "Unclassified"))
gabon_SS
dev.off()

pdf("/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/final_for_paper/codiversification/strain_sharing/files/strainsharing_Vietnam.pdf", width=6.75, height=5.75)
vietnam_SS <- comp50 %>%
  subset(name1_CurrCountry == "Vietnam" | name2_CurrCountry == "Vietnam") %>%
  subset(popANI >= 0.99999) %>%
  arrange(Level_new) %>%
  mutate(Level_new = factor(Level_new, levels=c("MIP", "Town/Village", "Region", "Country", "International"))) %>%
  ggplot(aes(x=Level_new, y=popANI)) +
    labs(title="Vietnam", x="Level", y = "popANI")+
    geom_boxplot(outlier.shape=NA, outlier.size=NA) +
    geom_jitter(size=3, width=0.3, height=0,aes(color=Phylum))+
    scale_y_continuous(limit = c(0.9999900, 1.0000000))+
    theme(plot.title = element_text(hjust = 0.5),
          text = element_text(family = "Helvetica"),
          axis.text=element_text(size=11),
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          panel.background = element_blank(), axis.line = element_line(colour = "black"),
          axis.title.x = element_text(margin = margin(t = 10)))+
    scale_x_discrete(name = "",
                   label = c("Mother-Child", "Town", "Region", "Country", "International"),
                   drop = FALSE)+
    scale_colour_manual(
        values = cols,
        breaks = c("p__actinobacteria", "p__bacteroidetes", "p__firmicutes", "p__proteobacteria", "p__spirochaetes","p__tenericutes", "unclassified"),
        labels = c("Actinobacteria", "Bacteroidetes", "Firmicutes", "Proteobacteria", "Spirochaetes","Tenericutes", "Unclassified"))
vietnam_SS
dev.off()

pdf("/ebio/abt3_projects/HUBIF_metagenomics/code/metagenomes/cophylogeny/instrain/final_for_paper/codiversification/strain_sharing/files/strainsharing_Germany.pdf", width=5.27, height=5.75)
germany_SS <- comp50 %>%
  subset(name1_CurrCountry == "Germany" | name2_CurrCountry == "Germany") %>%
  subset(popANI >= 0.99999) %>%
  arrange(Level_new) %>%
  mutate(Level_new = factor(Level_new, levels=c("MIP", "Town/Village", "International"))) %>%
  ggplot(aes(x=Level_new, y=popANI)) +
    labs(title="Germany", x="Level", y = "popANI")+
    geom_boxplot(outlier.shape=NA, outlier.size=NA) +
    geom_jitter(size=3, width=0.3, height=0,aes(color=Phylum))+
    scale_y_continuous(limit = c(0.9999900, 1.0000000))+
    theme(plot.title = element_text(hjust = 0.5),
          text = element_text(family = "Helvetica"),
          axis.text=element_text(size=11),
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          panel.background = element_blank(), axis.line = element_line(colour = "black"),
          axis.title.x = element_text(margin = margin(t = 10)))+
    scale_x_discrete(name = "",
                   label = c("Mother-Child", "Town", "International"),
                   drop = FALSE)+
    scale_colour_manual(
        values = cols,
        breaks = c("p__actinobacteria", "p__bacteroidetes", "p__firmicutes", "p__proteobacteria", "p__spirochaetes","p__tenericutes", "unclassified"),
        labels = c("Actinobacteria", "Bacteroidetes", "Firmicutes", "Proteobacteria", "Spirochaetes","Tenericutes", "Unclassified"))
germany_SS
dev.off()
```
# Summary

Figure 4A's components are written.

***

# SessionInfo

```{r sessionInfo}
sessionInfo()
```
